### DISCUSSION TITLE
Scope of globals() limited to current module?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16447] [General]
Scope of globals() limited to current module?  If I have a function or method name, and I want to call that method/function that happens to be defined in my 'main.py' file, but calling it is from a different module or class?

In main.py...
```
from Test import Testing

def Test():
    print('test')

Testing('Test')

```
In class Test:
```
class Test:
   def Testing(theMethod):
       globals()[theMethod]()
```
I have done this in regular method, but on micropython fails, looking at globals() it only shows method/functions in the current module.  I guess I could pass in a list of method/function pointers to 'Testing' and then test if in that list, then call the method in turn, but I wondered if there as a way to avoid this list of method/function pointers?

**Comment by andrewleech** [comment]
I think you should be able to do something like:
``` python
class Test:
   def Testing(theMethod):
        import main
        main.theMethod()
```

**Comment by Jibun-no-Kage** [comment]
Ah!  I will give that a shot... Thanks.  What I did so, was just create a function pointer dictionary,  Works, but not elegant at all.
